### PR TITLE
fix android resize

### PR DESCRIPTION
**Pull request by ahmdshrif**
### Fixed Issues

$ https://github.com/Expensify/App/issues/7591

### Tests

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

### QA Steps

1- Launch the app
2- Log in with any account
3- Tap on any user
4- Tap on plus button and add [attachment](https://user-images.githubusercontent.com/77761491/153068478-2f077913-23ff-46ee-85b6-4c1529343cd1.png)
5- Tap on Select from Gallery
6- Attach any picture from Gallery 
7- Tap Send
8- Tap on attachment again
-use different images sized like this one https://github.com/Expensify/App/pull/7660#issuecomment-1067032746
- [ ] verify that the image fits the screen
- [ ] verify that double tap to zoom works and there is no regression.
- [ ] Verify that no errors appear in the JS console

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [x] iOS
- [x] Android

#### Web

Screen Shot 2022-02-09 at 10 48 26 PM

#### Mobile Web

Screen Shot 2022-02-09 at 11 09 42 PM

#### Desktop

Screen Shot 2022-02-09 at 10 48 26 PM

#### iOS

Screen Shot 2022-03-24 at 7 29 19 AM

Screen Shot 2022-03-24 at 7 30 16 AM
Screen Shot 2022-03-24 at 7 30 32 AM

#### Android

Screen Shot 2022-03-24 at 6 46 27 AM
Screen Shot 2022-03-24 at 6 46 47 AM
Screen Shot 2022-03-24 at 6 48 21 AM
Screen Shot 2022-03-24 at 6 47 20 AM
Screen Shot 2022-03-24 at 6 48 35 AM

### CODE REVIEW

**Comment by ahmdshrif** [comment]
the conclusion here 
1-the old way of calculating the aspect ratio was not true so I do it the same way we use it on the web. 
2- to workaround the android issue we should only resize width or height in case of big images 
3- we will no longer need the resize at the end 
4- I add also resize image width to the max-width that android can handle without crashing. I add this code before and I delete width resize because we resize the image to container width on all cases. now we have a case we only resize the height. 
so we must add this safety resize for width.

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

File: src/components/ImageView/index.native.js [hunk 1]
```diff
@@ -64,22 +64,25 @@ class ImageView extends PureComponent {
...
-            // Fit the image to windowWidth
-            imageWidth = Math.round(this.props.windowWidth);
-            imageHeight = Math.round(imageWidth * aspectRatio);
+            const containerWidth = Math.round(this.props.windowWidth);
+            const containerHeight = Math.round(this.props.windowHeight - variables.contentHeaderHeight);
+
+            // const aspectRatio = imageHeight / imageWidth;
+            const aspectRatio = Math.min(containerHeight / imageHeight, containerWidth / imageWidth);
+
+            if (imageHeight < containerHeight && imageWidth < containerWidth) {
```
[the remark this answers is not in the document]
Comment (ahmdshrif) [reply src/components/ImageView/index.native.js:73]: @rushatgabhane  yes sure  i add it

File: src/components/ImageView/index.native.js [hunk 2]
```diff
@@ -64,22 +64,25 @@ class ImageView extends PureComponent {
...
+
+            const aspectRatio = Math.min(containerHeight / imageHeight, containerWidth / imageWidth);
+
+            // resize image to fit the screen dimensions if it small than screen else resize the smaller dimension only to avoid the resize issue on android
+            if (imageHeight < containerHeight && imageWidth < containerWidth) {
+                imageHeight *= aspectRatio;
+                imageWidth *= aspectRatio;
+            } else if (imageHeight > imageWidth) {
+                imageHeight *= aspectRatio;
+            } else {
```
Comment (rushatgabhane) [inline src/components/ImageView/index.native.js:78]: move this to a new line please

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@ahmdshrif please add the PR checklist to your post

I'm assuming tests and QA steps are the same. Could you please add more photos to tests (eg: https://github.com/Expensify/App/pull/7660#issuecomment-1067032746)?
And please write the tests more clearly. Something like 

- verify that the image fits the screen
- verify that double tap to zoom works and there is no regression.

File: src/components/ImageView/index.native.js (hunk 2 above)
Comment (ahmdshrif) [reply src/components/ImageView/index.native.js:78]: why move this to a new line? 
it will give lint issue

Comment (ahmdshrif) [reply src/components/ImageView/index.native.js:78]: @rushatgabhane why move this to a new line? 
it will give lint issue

**Comment by ahmdshrif** [comment]
@rushatgabhane  i edait the QA steps . 
Also i include screenshots with the image you mention.

File: src/components/ImageView/index.native.js (hunk 2 above)
Comment (rushatgabhane) [reply src/components/ImageView/index.native.js:78]: TIL we have this lint rule. Thanks @ahmdshrif

**Comment by rushatgabhane** [comment]
> please add the PR checklist to your post

@ahmdshrif I think you missed this comment

**Comment by ahmdshrif** [comment]
@rushatgabhane  sorry I miss this.  but add it now.

**Review by rushatgabhane — APPROVED** [review]
@AndrewGable LGTM!

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

**Comment by ahmdshrif** [comment]
@rushatgabhane now I add screenshots.

**Review by AndrewGable — APPROVED** [review]

**Pull request merged**